content/Kotlin Scope Functions - Amaury Ricardo.md

```diff
@@ -0,0 +1,107 @@
+## Scope functions in Kotlin
+![Kotlin](images/kotlin.png)
+
+### Objectives.
+. Work with the same context.
+. Write clean, readable and easy to maintain code
+
+### Scope functions:
+. run
+. let
+. apply
+. also
+. with
+
+### How and where to use:
+1. run:
+Receive 'this' as an instance of the current context and return a final value, should be used to execute actions over the same context.
+```
+val dog = Dog()
+val result = dog.run { //this as instance of the object dog
+	jump() // same as dog.jump() also you can use this.jump()
+	sit()
+  eat()
+	sleep() // the final return value
+}
+```
+Without "run" the code looks like:
+```
+dog.jump()
+dog.sit()
+dog.eat()
+val resutl = dog.sleep()
+```
+
+2. let:
+Receive 'it' as an instance of the current object and return a final value, should be used to use the object context several times
```

Comment:
Also, I would use 'helpful for this case...' or 'It is recommended for ...' instead of 'Should be used'. This applies to all other method descriptions.